src/Stratis.Bitcoin.Features.Interop/InteropBehavior.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Numerics;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Features.Interop.EthereumClient;
+using Stratis.Bitcoin.Features.Interop.Payloads;
+using Stratis.Bitcoin.Features.PoA;
+using Stratis.Bitcoin.P2P.Peer;
+using Stratis.Bitcoin.P2P.Protocol;
+using Stratis.Bitcoin.P2P.Protocol.Behaviors;
+using Stratis.Bitcoin.Utilities;
+using TracerAttributes;
+
+namespace Stratis.Bitcoin.Features.Interop
+{
+    public class InteropBehavior : NetworkPeerBehavior
+    {
+        private readonly ILoggerFactory loggerFactory;
+
+        private readonly ILogger logger;
+
+        private readonly Network network;
+
+        private readonly IFederationManager federationManager;
+
+        private readonly IInteropTransactionManager interopTransactionManager;
+
+        private readonly IEthereumClientBase ethereumClientBase;
+
+        public InteropBehavior(
+            ILoggerFactory loggerFactory,
+            Network network,
+            IFederationManager federationManager,
+            IInteropTransactionManager interopTransactionManager,
+            IEthereumClientBase ethereumClientBase)
+        {
+            Guard.NotNull(loggerFactory, nameof(loggerFactory));
+            Guard.NotNull(network, nameof(network));
+            Guard.NotNull(federationManager, nameof(federationManager));
+            Guard.NotNull(interopTransactionManager, nameof(interopTransactionManager));
+            Guard.NotNull(ethereumClientBase, nameof(ethereumClientBase));
+
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+            this.loggerFactory = loggerFactory;
+            this.network = network;
+            this.federationManager = federationManager;
+            this.interopTransactionManager = interopTransactionManager;
+            this.ethereumClientBase = ethereumClientBase;
+        }
+
+        [NoTrace]
```

Comment:
fody was removed from codebase according to @fassadlr , so this tag is not needed